File: EPI/9.2/swift/9.2_test.swift
import Darwin

let tests: [(expression: String, result: Int)] = [
    ("1", 1),
    ("-1", -1),
    ("2,3,+", 5),
    ("5,2,-", 3),
    ("2,3,*", 6),
    ("6,2,/", 3),
    ("1,3,+,5,7,8,+,*,-", -71),
]

for test in tests {
    let result = evaluate(test.expression)!
    guard result == test.result else {
        print("For expression \(test.expression), " +
            "expected evaluated result to be \(test.result), but was \(result)")
        exit(1)
    }
}

File: EPI/9.2/typescript/9.2_test.ts
import evaluate from './9.2';

declare function require(name: string): any;
const assert = require('assert');
// const inspect = (v: any) => require('util').inspect(v, {depth: null});

const tests: {rpn: string, value: number}[] = [
    {rpn: '1', value: 1},
    {rpn: '-1', value: -1},
    {rpn: '2,3,+', value: 5},
    {rpn: '5,2,-', value: 3},
    {rpn: '2,3,*', value: 6},
    {rpn: '6,2,/', value: 3},
    {rpn: '1,3,+,5,7,8,+,*,-', value: -71},
];

for (const test of tests) {
    const actual = evaluate(test.rpn);
    assert.strictEqual(actual, test.value, `For RPN expression ${test.rpn}, ` +
        `expected value to be ${test.value}, but was ${actual}`);
}
